torch/onnx/_type_utils.py

```diff
@@ -108,36 +113,96 @@ def from_name(
     @classmethod
     @_beartype.beartype
     def from_dtype(cls, dtype: torch.dtype) -> JitScalarType:
-        """Convert a torch dtype to ScalarType."""
+        """Convert a torch dtype to JitScalarType.
+
+        NB: DO NOT USE this API if dtype is comes from a torch._C.Value.type().dtype() call because
+            "RuntimeError: INTERNAL ASSERT FAILED at "../aten/src/ATen/core/jit_type_base.h" can
+            be raised in some scenarios. Instead use from_value API which handles all cases
+
+        Args:
+            dtype: A torch.dtype to create a JitScalarType from
+
+        Returns:
+            JitScalarType
+
+        Raises:
+            ValueError: if dtype is not a valid torch.dtype or if it is None.
+        """
         if dtype not in _DTYPE_TO_SCALAR_TYPE:
             raise ValueError(f"Unknown dtype: {dtype}")
         return _DTYPE_TO_SCALAR_TYPE[dtype]
 
+    # TODO: `value: torch._C.Value` type annotation raises (CI machines only):
+    #       NameError: name '_C' is not defined
+    # TODO: Return type Optional[JitScalarType] or Union[JitScalarType, None] raises (CI machines only):
+    #       NameError: name 'JitScalarType' is not defined
+    # In both cases, using typing.TypeVar or string literals didn't work
+    @classmethod
+    @_beartype.beartype
+    def from_value(cls, value, allow_list_type: bool = True, raises: bool = True):
```

Comment:
I wonder if we can make the API more simple by taking out `allow_list_type: bool = True, raises: bool = True`?

It looks like `allow_list_type` is never set to False in our usage and we can factor out None checking on `value` to the caller?

Reply:
`raises` can be removed with some refactoring. Will do, thanks

You actually caught a bug because `allow_list_type` was supposed to be used by `_is_fp`

However, I will try to remove it too and see if there is any side-effect. I was concerned about always allowing list types when looking for the underlying scalar type, but maybe it is ok.

Reply:
It seems to have worked. Will keep these flags out. Thanks